tests/heka/test_message_parser.py

```diff
@@ -1,87 +1,121 @@
-            assert set(r.keys()) == top_keys
-            assert set(r["payload"].keys()) == payload_keys
+        if "gzip" in heka_format:
+            o = streaming_gzip_wrapper(o)
+        try:
+            for r, b in message_parser.unpack(o, try_snappy=try_snappy, strict=strict):
+                j = json.loads(r.message.payload)
+                assert count == j["seq"]
+                count += 1
+        except DecodeError:
+            threw_exception = True
 
-    filename = "{}/test_telemetry_snappy.heka".format(data_dir)
-    with open(filename, "rb") as o:
-        for r in message_parser.parse_heka_message(o):
-            assert set(r.keys()) == top_keys
-            assert set(r["payload"].keys()) == payload_keys
+    assert count == expected_count
+    assert threw_exception == expected_exception
+
+
+@pytest.mark.parametrize("heka_format", ["snappy", "gzip"])
+def test_telemetry(data_dir, heka_format):
+    filename = "{}/test_telemetry_{}.heka".format(data_dir, heka_format)
+    reference_filename = filename + '.json'
+
+    # enable this to regenerate the expected json representation of the ping
+    if False:
+        with open(filename, "rb") as f:
+            if "gzip" in heka_format:
+                f = streaming_gzip_wrapper(f)
+            # deep copy the parsed message so lazy-parsed json gets vivified
+            msg = copy.deepcopy(message_parser.parse_heka_message(f).next())
+            open(reference_filename, 'w').write(json.dumps(msg, indent=4,
+                                                           sort_keys=True))
+
+    reference = json.load(open(reference_filename))
+    with open(filename, "rb") as f:
+        if "gzip" in heka_format:
+            f = streaming_gzip_wrapper(f)
+        # deep copy the parsed message so lazy-parsed json gets vivified
+        msg = copy.deepcopy(message_parser.parse_heka_message(f).next())
+        assert msg == reference
 
 
 def test_invalid_utf8(data_dir):
     filename = "{}/test_invalid_utf8.heka".format(data_dir)
     with open(filename, "rb") as o:
         for r in message_parser.parse_heka_message(o):
             assert(u'\ufffd' in r['info']['adapterDescription'])
+
+
+def test_json_fallback():
+    # ujson can't handle values which are too large
+    # validate that we fallback correctly to standard json for this
+    # case (https://github.com/esnme/ultrajson/issues/252)
+    TOO_BIG = 272757895493505930073807329622695606794392
+
+    # first assert that the exception still exists (so we can remove the
+    # fallback in the future)
+    with pytest.raises(ValueError):
+        ujson.loads(str(TOO_BIG))
+
+    # now assert that the fallback to standard json works
+    from moztelemetry.heka.message_parser import _parse_json
+    assert TOO_BIG == _parse_json(str(TOO_BIG))
+
+
+def test_lazy_parsing(data_dir, monkeypatch):
+    from moztelemetry.heka.message_parser import _parse_json
```

Comment:
Could you combine these imports into one? importing `message_parser` should make it work for both MagicMock (message_parser._parse_json) and monkeypatch